CDTDatastore/touchdb/TD_Database.m

```diff
@@ -522,18 +531,30 @@ INSERT INTO INFO (key, value) VALUES ('publicUUID',  '%@')",
     }];
     
     if (result) {
-        _open = YES;
+        self.open = YES;
         return YES;
     } else {
         return NO;
     }
 }
 
 #pragma mark Closing and deleting database
-
 - (BOOL)close
 {
-    if (![self isOpen]) return NO;
+
+    __block BOOL result;
+    dispatch_sync(self.queue, ^{
+        result = [self closeOnQueue];
+    });
+
+    return result;
+}
+
+- (BOOL) closeOnQueue
```

Comment:
I don't like this method name as the `OnQueue` suffix implies it queues it whereas this method itself doesn't use a queue but you intend that it should be called from something that does queue it.  I think I'd call it `closeInternal` or `closePrivate` (unless you can think of a better name) and maybe add a comment suggesting it should only be called when wrapped in a block on the dispatch queue.